File: Project_Sourcecode_java/src/main/java/edu/iit/cs586/project/state/Ready.java
package edu.iit.cs586.project.state;

import edu.iit.cs586.project.MDA_EFSM;
import edu.iit.cs586.project.output.OP;

/**
 * @author CharmingLee <charminglee@jieztech.com>
 * @date 2022/5/1 22:28
 */
public class Ready extends State{
    public Ready(MDA_EFSM m) {
        super(m);
    }

    @Override
    public String getName() {
        return "Ready";
    }

    @Override
    public void logout() {
        this.m.changeState(1);
    }

    @Override
    public void withdraw() {
        m.p.makeWithdraw();
        this.m.changeState(4);
    }

    @Override
    public void balance() {
        m.p.displayBalance();
    }

    @Override
    public void lock() {
        this.m.changeState(6);
    }

    @Override
    public void incorrectLock() {
        m.p.IncorrectLockMsg();
    }

    @Override
    public void deposit() {
        m.p.makeDeposit();
    }

    @Override
    public void suspend() {
        m.changeState(7);
    }

    @Override
    public void noFunds() {
        m.p.noFundsMsg();
    }
}

File: Project_Sourcecode_go/state/Ready.go
package state

type Ready struct {
	State
}

func (s *Ready) Logout() {
	s.M.ChangeState(1)
}
func (s *Ready) NoFunds() {
	s.M.P.NoFundsMsg()
}
func (s *Ready) Suspend() {
	s.M.ChangeState(7)
}
func (s *Ready) Deposit() {
	s.M.P.MakeDeposit()
}
func (s *Ready) IncorrectLock() {
	s.M.P.IncorrectLockMsg()
}
func (s *Ready) Lock() {
	s.M.ChangeState(6)
}
func (s *Ready) Withdraw() {
	s.M.P.MakeWithdraw()
	s.M.ChangeState(4)
}
func (s *Ready) Balance() {
	s.M.P.DisplayBalance()
}

func (s *Ready) GetName() string {
	return "Ready"
}
